### PR TITLE
create thumbnail with appropriate height and width

### PR DESCRIPTION
**Pull request by parveshneedhoo**
Generate thumbnail with appropriate height and width for ios.

### CODE REVIEW

File: src/android/Thumbnails.java [hunk 1]
```diff
@@ -157,4 +157,4 @@ public static class Options {
         public Options() {
         }
     }
-}
+}
```
Comment (dinitri) [inline src/android/Thumbnails.java]:
```suggestion
}

```

**Review by dinitri — COMMENTED** [review]
new line

File: src/ios/Thumbnail.m [hunk 2]
```diff
@@ -85,25 +85,33 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
-    NSDictionary *imageOptions = @{
-                                   (NSString const *)kCGImageSourceCreateThumbnailFromImageIfAbsent : (NSNumber const *)kCFBooleanTrue,
-                                   (NSString const *)kCGImageSourceThumbnailMaxPixelSize            : @(maxPixelSize),
-                                   (NSString const *)kCGImageSourceCreateThumbnailWithTransform     : (NSNumber const *)kCFBooleanTrue,
-                                   (NSString const *)kCGImageSourceCreateThumbnailFromImageAlways   : (NSNumber const *)kCFBooleanTrue
-                                   };
-    CGImageRef thumbnail = CGImageSourceCreateThumbnailAtIndex(imageSource, 0, (__bridge CFDictionaryRef)imageOptions);
-    CFRelease(imageSource);
+    NSData *data = [NSData dataWithContentsOfURL:URL];
+    UIImage* image = [[UIImage alloc]initWithData:data];
```
Comment (zfir) [inline src/ios/Thumbnail.m]:
```suggestion
    UIImage* image = [[UIImage alloc] initWithData:data];
```

File: src/ios/Thumbnail.m [hunk 3]
```diff
@@ -85,25 +85,33 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
+        ratio = MAX(image.size.width/maxPointSize, image.size.height/maxPointSize);
+    } else {
+        ratio = MIN(image.size.width/maxPointSize, image.size.height/maxPointSize);
+    }
+    
+    if (@available(iOS 15.0, *)) {
+        double thumbnailHeight = (image.size.height/ratio) / image.scale;
+        double thumbnailWidth = (image.size.width/ratio) / image.scale;
+        CGSize thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
+        UIImage *newImage = [image imageByPreparingThumbnailOfSize: thumbnailSize];
```
Comment (zfir) [inline src/ios/Thumbnail.m]:
```suggestion
        return [image imageByPreparingThumbnailOfSize: thumbnailSize];
```

File: src/ios/Thumbnail.m [hunk 4]
```diff
@@ -85,25 +85,33 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
+    double maxPointSize = maxPixelSize / (double) (image.scale);
+    double ratio;
+    if ((image.size.width / image.size.height) > 1) {
+        ratio = MAX(image.size.width/maxPointSize, image.size.height/maxPointSize);
+    } else {
+        ratio = MIN(image.size.width/maxPointSize, image.size.height/maxPointSize);
+    }
+    
+    if (@available(iOS 15.0, *)) {
+        double thumbnailHeight = (image.size.height/ratio) / image.scale;
```
Comment (zfir) [inline src/ios/Thumbnail.m]:
```suggestion
        double thumbnailHeight = (image.size.height / ratio) / image.scale;
```
do for the other lines also

File: src/ios/Thumbnail.m [hunk 5]
```diff
@@ -85,25 +85,33 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
-
-    return result;
+    double maxPointSize = maxPixelSize / (double) (image.scale);
+    double ratio;
+    if ((image.size.width / image.size.height) > 1) {
+        ratio = MAX(image.size.width/maxPointSize, image.size.height/maxPointSize);
+    } else {
+        ratio = MIN(image.size.width/maxPointSize, image.size.height/maxPointSize);
+    }
+
```
Comment (zfir) [inline src/ios/Thumbnail.m:101]:
```suggestion
    double thumbnailHeight;
    double thumbnailWidth;
    CGSize thumbnailSize;
    
```

File: src/ios/Thumbnail.m [hunk 6]
```diff
@@ -85,25 +85,33 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
+    if (@available(iOS 15.0, *)) {
+        double thumbnailHeight = (image.size.height/ratio) / image.scale;
+        double thumbnailWidth = (image.size.width/ratio) / image.scale;
+        CGSize thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
+        UIImage *newImage = [image imageByPreparingThumbnailOfSize: thumbnailSize];
+        return newImage;
+    } else {
+        double thumbnailHeight = (image.size.height/ratio) / UIScreen.mainScreen.scale;
+        double thumbnailWidth = (image.size.width/ratio) / UIScreen.mainScreen.scale;
+        CGSize thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
```
Comment (zfir) [inline src/ios/Thumbnail.m]:
```suggestion
        thumbnailHeight = (image.size.height/ratio) / UIScreen.mainScreen.scale;
        thumbnailWidth = (image.size.width/ratio) / UIScreen.mainScreen.scale;
        thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
```

File: src/ios/Thumbnail.m [hunk 7]
```diff
@@ -85,25 +85,35 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
+    
+    if (@available(iOS 15.0, *)) {
+        thumbnailHeight = (image.size.height / ratio) / image.scale;
+        thumbnailWidth = (image.size.width / ratio) / image.scale;
+        thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
+        return [image imageByPreparingThumbnailOfSize: thumbnailSize];
+    } else {
+        thumbnailHeight = (image.size.height / ratio) / UIScreen.mainScreen.scale;
+        thumbnailWidth = (image.size.width / ratio) / UIScreen.mainScreen.scale;
+        thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
```
Comment (zfir) [inline src/ios/Thumbnail.m:110]:
```suggestion
```

File: src/ios/Thumbnail.m [hunk 8]
```diff
@@ -85,25 +85,35 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
+    if ((image.size.width / image.size.height) > 1) {
+        ratio = MAX(image.size.width / maxPointSize, image.size.height / maxPointSize);
+    } else {
+        ratio = MIN(image.size.width / maxPointSize, image.size.height / maxPointSize);
+    }
+    
+    if (@available(iOS 15.0, *)) {
+        thumbnailHeight = (image.size.height / ratio) / image.scale;
+        thumbnailWidth = (image.size.width / ratio) / image.scale;
+        thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
```
Comment (zfir) [inline src/ios/Thumbnail.m:105]:
```suggestion
```

File: src/ios/Thumbnail.m [hunk 9]
```diff
@@ -85,25 +85,35 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
 {
-    CGImageSourceRef imageSource = CGImageSourceCreateWithURL((__bridge CFURLRef)URL, NULL);
-    if(imageSource == NULL) {
-        NSLog(@"Can not read from source: %@", URL);
-        return NULL;
+    NSData *data = [NSData dataWithContentsOfURL:URL];
+    UIImage* image = [[UIImage alloc] initWithData:data];
+    double thumbnailHeight;
+    double thumbnailWidth;
+    CGSize thumbnailSize;
```
Comment (zfir) [inline src/ios/Thumbnail.m:92]:
```suggestion
    CGSize thumbnailSize = CGSizeMake(thumbnailWidth, thumbnailHeight);
```

File: src/ios/Thumbnail.m [hunk 10]
```diff
@@ -85,25 +85,35 @@ + (void) thumbnail:(NSString *)imageURL size:(CGFloat)maxSize toURL:(NSString *)
...
+    double maxPointSize = maxPixelSize / (double) (image.scale);
+    double ratio;
+
+    if ((image.size.width / image.size.height) > 1) {
```
Comment (zfir) [inline src/ios/Thumbnail.m]:
```suggestion
    double maxPointSize = maxPixelSize / (double) (image.scale);
    
    double ratio;
    if ((image.size.width / image.size.height) > 1) {
```

**Pull request merged**